java-extensions/hadoop-ext/src/main/java/org/apache/hadoop/fs/FileSystem.java

```diff
@@ -3759,6 +3753,15 @@ private static FileSystem createFileSystem(URI uri, Configuration conf)
         }
     }
 
+    private static FileSystem createFileSystem(URI uri, Configuration conf)
+            throws IOException {
+        HadoopExt.getInstance().rewriteConfiguration(conf);
+        UserGroupInformation ugi = HadoopExt.getInstance().getHDFSUGI(conf);
+        FileSystem fs = HadoopExt.getInstance().doAs(ugi, () -> createFileSystemInternal(uri, conf));
+        FileSystem fs2 = HadoopExt.getInstance().bindUGIToFileSystem(fs, ugi);
```

Comment:
Can you add some comment in the code to explain why you create fs2?

Reply:
Dear, actually there is no need to bind ugi to fs now. 

I've removed it.